discovery_engine_core/core/src/storage/sqlite.rs

```diff
@@ -84,113 +92,301 @@ impl Storage for SqliteStorage {
+        #[derive(sqlx::FromRow)]
+        struct _ApiDocumentView {
+            document: Uuid,
+            title: String,
+            snippet: String,
+            topic: String,
+            url: String,
+            image: Option<String>,
+            date_published: NaiveDateTime,
+            source: String,
+            market: String,
+            domain_rank: u32,
+            score: f32,
+            user_reaction: Option<u32>,
+            in_batch_index: u32,
+        }
+
+        let mut con = self.pool.acquire().await.map_err(DatabaseError::Sql)?;
+        sqlx::query_as::<_, _ApiDocumentView>(
+            "SELECT nr.document, nr.title, nr.snippet, nr.topic, nr.url, nr.image,
+            nr.datePublished, nr.source, nr.market, nc.domainRank, nc.score, ur.userReaction,
+            po.inBatchIndex
+            FROM NewsResource as nr, NewscatcherData as nc, UserReaction as ur,
+            FeedDocuments as fd, PresentationOrdering as po
+            ON fd.document = nr.document, fd.document = nc.document,
+            fd.document = ur.document, fd.document = po.document
+            ORDERED BY po.timestamp, po.inBatchIndex ASC;",
+        )
+        .fetch_all(&mut con)
+        .await
+        .map_err(DatabaseError::Sql)?
+        .into_iter()
+        .map(|doc| {
+            let url = Url::parse(&doc.url).map_err(|e| DatabaseError::TypeConversion(e.into()))?;
+            let image = doc
+                .image
+                .map(|url| Url::parse(&url).map_err(|e| DatabaseError::TypeConversion(e.into())))
+                .transpose()?;
+            let (lang_code, country_code) = doc.market.split_at(2);
+            let market = Market {
+                lang_code: lang_code.to_string(),
+                country_code: country_code.to_string(),
+            };
+
+            let news_resource = NewsResource {
+                title: doc.title,
+                snippet: doc.snippet,
+                topic: doc.topic,
+                url,
+                image,
+                date_published: doc.date_published,
+                source: doc.source,
+                market,
+            };
+            let newscatcher_data = NewscatcherData {
+                domain_rank: doc.domain_rank as usize,
+                score: Some(doc.score),
+            };
+            let user_reacted = doc.user_reaction.and_then(FromPrimitive::from_u32);
+
+            Ok(ApiDocumentView {
+                document: document::Id::from(doc.document),
+                news_resource,
+                newscatcher_data,
+                user_reacted,
+                in_batch_index: doc.in_batch_index as usize,
             })
-            .collect()
+        })
+        .collect()
+    }
+
+    async fn store_documents(
+        &self,
+        documents: &[document::Document],
+    ) -> Result<(), Self::FeedScopeError> {
+        if documents.is_empty() {
+            return Ok(());
+        }
+
+        // const BIND_LIMIT: usize = 32766;
```

Comment:
Do we need such check or is it unrealistic that we will ever have so many documents?

Reply:
We are not cleaning anything so it is currently possible. Is this limit only for 32bit bindings or every binding?
Anyway, panic is not acceptable in this case, we can take the first `BIND_LIMIT` documents and discard the others.

Reply:
> We are not cleaning anything so it is currently possible. Is this limit only for 32bit bindings or every binding?

looks like for every binding. but the version is important SQLite prior to 3.32.0 only allowed [999](https://docs.rs/sqlx-core/latest/sqlx_core/query_builder/struct.QueryBuilder.html#note-database-specific-limits)

Reply:
Note that this is the bind limit, not db size limit. 

I.e. in this context it's the max number of documents we can add to the feed at once.

We can ignore it here I think, even if we are on a older sqlite version.


(but shouldn't panic)

Reply:
I implemented it so that we only take the first (BIND_LIMIT / number of fields for the largest tuple) documents